src/plugins/query_enhancements/public/query_assist/components/query_assist_summary.tsx

```diff
@@ -0,0 +1,286 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+import {
+  EuiSplitPanel,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiText,
+  EuiIcon,
+  EuiIconTip,
+  EuiSmallButtonIcon,
+  EuiSpacer,
+  EuiCopy,
+} from '@elastic/eui';
+
+import React, { useEffect, useState } from 'react';
+import { i18n } from '@osd/i18n';
+import { IDataFrame } from 'src/plugins/data/common';
+import { v4 as uuidv4 } from 'uuid';
+import { isEmpty } from 'lodash';
+import { merge, of } from 'rxjs';
+import { filter, distinctUntilChanged, mergeMap } from 'rxjs/operators';
+import { HttpSetup } from 'opensearch-dashboards/public';
+import { useQueryAssist } from '../hooks';
+import { DataPublicPluginSetup, QueryEditorExtensionDependencies } from '../../../../data/public';
+import { UsageCollectionSetup } from '../../../../usage_collection/public';
+import { CoreSetup } from '../../../../../core/public';
+import sparkleHollowSvg from '../../assets/sparkle_hollow.svg';
+
+export interface QueryContext {
+  question: string;
+  query: string;
+  queryResults: any;
+}
+
+export interface QueryAssistSummaryProps {
+  data: DataPublicPluginSetup;
+  http: HttpSetup;
+  usageCollection: UsageCollectionSetup;
+  dependencies: QueryEditorExtensionDependencies;
+  core: CoreSetup;
+}
+
+export const QueryAssistSummary: React.FC<QueryAssistSummaryProps> = (props) => {
+  const [summary, setSummary] = useState(null); // store fetched data
+  const [loading, setLoading] = useState(false); // track loading state
+  const [queryContext, setQueryContext] = useState<QueryContext | undefined>(undefined);
+  const [feedback, setFeedback] = useState(false);
+  const [isAssistantEnabledByCapability, setIsAssistantEnabledByCapability] = useState(false); // track loading state
+  const { question$, isQueryAssistCollapsed } = useQueryAssist();
+  const METRIC_APP = `query-assistant`;
+  const afterFeedbackTip =
+    'Thank you for the feedback. Try again by adjusting your question so that I have the opportunity to better assist you.';
+
+  const sampleSize = 10;
```

Comment:
if you have a services.uiSettings avaiable we can consider adding a new Advanced Setting that lets this be configurable

Reply:
Agreed with you to make to configurable, but I'll defer this to future implementation. We might need to change the way to choose sampling data to summarize.